src/de.rs

```diff
@@ -27,34 +14,28 @@ where
 }
 
 /// A structure that deserializes Python objects into Rust values
-pub struct Depythonizer<'py> {
-    input: Bound<'py, PyAny>,
+pub struct Depythonizer<'py, 'bound> {
+    input: &'bound Bound<'py, PyAny>,
 }
 
-impl<'py> Depythonizer<'py> {
-    /// Create a deserializer from a Python object
-    #[deprecated(
-        since = "0.21.0",
-        note = "will be replaced by `Depythonizer::from_object_bound` in a future version"
-    )]
-    pub fn from_object(input: &'py PyAny) -> Self {
-        Self::from_object_bound(input.as_borrowed().to_owned())
-    }
-
+impl<'py, 'bound> Depythonizer<'py, 'bound> {
     /// Create a deserializer from a Python object
-    pub fn from_object_bound(input: Bound<'py, PyAny>) -> Self {
+    pub fn from_object_bound(input: &'bound Bound<'py, PyAny>) -> Self {
```

Comment:
Similarly let's deprecate `from_object_bound` and replace `from_object` instead.

Reply:
I can't implement for `Bound` the item would be dropped. Anyway, just add a lifetime parameter to `Depythonizer` struct is breaking change. What we can do is have two `Depythonizer` struct one with the ref the other without.